src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.Validation.cs

```diff
@@ -117,22 +119,35 @@ internal static unsafe partial class Utf8Utility
                         // the alignment check consumes at most a single DWORD.)
 
                         byte* pInputBufferFinalPosAtWhichCanSafelyLoop = pFinalPosWhereCanReadDWordFromInputBuffer - 3 * sizeof(uint); // can safely read 4 DWORDs here
-                        uint mask;
+                        int trailingZeroCount;
```

Comment:
Perf: prefer `nuint trailingZeroCount`. Avoids a signed extension later in the method.

Reply:
There is no `BitConverter.TrailingZeroCount` overload that returns a `uint` / `nuint`. The invocation below passes a `ulong` mask and returns a `ulong`. Would it be safe to cast the returned value into a `uint`?

Reply:
Ok I think the maximum possible numbers of zeroes that can be counted from the ulong would probably not go beyond the maximum possible number that a uint can hold.